packages/babel-helper-module-transforms/src/index.js

```diff
@@ -166,26 +171,35 @@ function buildESModuleHeader(
 /**
  * Create a re-export initialization loop for a specific imported namespace.
  */
-function buildNamespaceReexport(metadata, namespace) {
-  return template.statement`
-    Object.keys(NAMESPACE).forEach(function(key) {
-      if (key === "default" || key === "__esModule") return;
-      VERIFY_NAME_LIST;
-
-      Object.defineProperty(EXPORTS, key, {
-        enumerable: true,
-        get: function() {
-          return NAMESPACE[key];
-        },
-      });
-    });
-  `({
+function buildNamespaceReexport(metadata, namespace, loose) {
+  return (loose
+    ? template.statement`
```

Comment:
maybe you could hoist those 2 templates instead of inlining them? aint sure how templates work and if once created template wont reuse nodes or something on subsequent calls

Reply:
I thought logan's changes were to fix that issue of having to do that https://github.com/babel/babel/pull/6492, should probably go into a changelog lol but there's so much

Reply:
Wouldn't this still instantiate the template every time new? Or does babel-template now internally cache templates and not reparse them? I tried looking at the code, but couldn'T figure it out myself. @loganfsmyth

Reply:
Using tagged templates will cache them inline so we don't need to stick them all at the top anymore.